### ISSUE TITLE
OpenBSD/FreeBSD does not work in TUN mode due to PI header

### ISSUE THREAD

**Issue by neilalexander** [yggdrasil-network/yggdrasil-go#31]
OpenBSD does not have the ability to disable the 4-byte protocol information (PI) header from each packet to/from the TUN adapter.

This causes Yggdrasil to crash due to a `write` error on the TUN, as we do not currently account for this anywhere.

TAP mode works fine on OpenBSD as the PI header is not included there.

**Comment by Arceliar** [comment]
I guess maybe we could add a function to take a `[]byte` of the packet, and return a `[]byte` with any platform-specific changes needed. The default implementation could return the `[]byte` unchanged. The OpenBSD implementation would strip the PI header on reads, and add a PI header on writes. However, having to do this in our code for only OpenBSD is pretty ugly, so maybe it would make more sense to do it in our fork of water (and talk to the upstream devs about it)?

**Comment by McL0v1n** [comment]
@neilalexander Is this still applicable due to the change to wireguard?

**Comment by z3bra** [comment]
I got latest build (v0.3.14) to work in TUN mode on OpenBSD, using wireguard. I changed to following options in the default configuration:

    ifName: tun3 # using "/dev/tun3" doesn't work, as wireguard checks the name against the regex "tun[0-9]*"
    IfMTU: 1500 # wireguard seems unable to change the MTU on the tun interface, so I had to use the default value of 1500

Other than that, everything seems to work, as I could access a few services on the Ygg network. I managed to run the daemon as an unprivileged user as well, by changing ownership of `/dev/tun3` to the daemon user.
